framework/src/postprocessors/PseudoTimestep.C

```diff
@@ -0,0 +1,190 @@
+
+InputParameters
+PseudoTimestep::validParams()
+{
+  InputParameters params = GeneralPostprocessor::validParams();
+
+  MooseEnum method("SER=1 EXP=2 RDM=3");
+  params.addRequiredParam<MooseEnum>("method", method, "The method used for CFL scaling");
+  params.addParam<Real>("initial_dt", "Initial timestep");
+  params.addParam<Real>("alpha", 1.0, "The parameter alpha used in the scaling of the timestep");
+  params.addParam<Real>("max_dt", "The largest timestp allowed");
+  params.addParam<unsigned int>(
+      "iterations_window",
+      1.0,
+      "For how many iterations should the CFL be tracked (only applies to the SER method)");
+  // Because this post-processor is meant to be used with PostprocessorDT, it
+  // should be executed on initial (not included by default) and timestep end.
+  params.set<ExecFlagEnum>("execute_on") = {EXEC_TIMESTEP_END, EXEC_INITIAL, EXEC_TIMESTEP_BEGIN};
+  params.suppressParameter<ExecFlagEnum>("execute_on");
+
+  params.addClassDescription("Computes a time step size to accelerate iterations");
+
+  return params;
+}
+
+PseudoTimestep::PseudoTimestep(const InputParameters & parameters)
+  : GeneralPostprocessor(parameters),
+    _method(getParam<MooseEnum>("method")),
+    _initial_dt(getParam<Real>("initial_dt")),
+    _alpha(getParam<Real>("alpha")),
+    _max_dt(getParam<Real>("max_dt")),
+    _iterations_window(getParam<unsigned int>("iterations_window"))
+{
+}
+
+void
+PseudoTimestep::initialize()
+{
+  // start with the max
+  _dt = std::numeric_limits<Real>::max();
+}
+
+Real
+PseudoTimestep::getValue() const
+{
+  return _dt;
+}
+
+Real
+PseudoTimestep::current_residual_norm() const
+{
+  NonlinearSystemBase & nl = _fe_problem.getNonlinearSystemBase();
+  Real var_res;
+  Real res_norm = 0.0;
+  for (unsigned int var_num = 0; var_num < nl.system().n_vars(); var_num++)
+  {
+    var_res = nl.system().calculate_norm(nl.getResidualNonTimeVector(), var_num, DISCRETE_L2);
+    res_norm = res_norm + std::pow(var_res, 2);
+  }
+  res_norm = std::sqrt(res_norm);
+  return res_norm;
+}
+
+void
+PseudoTimestep::output_pseudo_timestep(Real & curr_dt, const ExecFlagType & exec_type)
+{
+  unsigned int curr_step = _fe_problem.timeStep();
+
+  unsigned int steps_size = _iterations_step_sequence.size();
+  unsigned int res_size = _residual_norms_sequence.size();
+
+  Moose::out << std::endl;
+  Moose::out << "***********************************************" << std::endl;
+  Moose::out << "Current step " << curr_step << " executed at " << _current_execute_flag
+             << " current timestep " << curr_dt << std::endl;
+
+  Moose::out << "Iterations vector is of size " << steps_size << " , and residual norms is of size "
+             << res_size << std::endl;
+  Moose::out << "Current dt: " << _iterations_step_sequence[steps_size - 1]
+             << " and residual: " << _residual_norms_sequence[res_size - 1] << std::endl;
+
+  Moose::out << std::endl;
+}
+
+Real
+PseudoTimestep::timestep_SER()
```

Comment:
```suggestion
PseudoTimestep::timestep_SER() const
```

Reply:
Note taken, but I'm not eager to work with consts while work is in progress in case some variable needs to be recast. I will resolve the conversations related to this.

Reply:
These functions can be made const.